jsonrpc/eth_endpoint.go

```diff
@@ -802,3 +802,78 @@ func (e *Eth) MaxPriorityFeePerGas() (interface{}, error) {
 
 	return argBigPtr(priorityFee), nil
 }
+func (e *Eth) FeeHistory(blockCount uint64, newestBlock uint64, rewardPercentiles []float64) (interface{}, error) {
+	type feeHistoryResult struct {
+		OldestBlock   argUint64
+		BaseFeePerGas []argUint64
+		GasUsedRatio  []argUint64
+		Reward        [][]argUint64
+	}
+
+	// Retrieve oldestBlock, baseFeePerGas, gasUsedRatio, and reward synchronously
+	oldestBlock, baseFeePerGas, gasUsedRatio, reward, err := e.store.FeeHistory(blockCount, newestBlock, rewardPercentiles)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create channels to receive the processed slices asynchronously
+	baseFeePerGasCh := make(chan []argUint64)
+	gasUsedRatioCh := make(chan []argUint64)
+	rewardCh := make(chan [][]argUint64)
+
+	// Process baseFeePerGas asynchronously
+	go func() {
+		baseFeePerGasCh <- convertToArgUint64Slice(*baseFeePerGas)
+	}()
+
+	// Process gasUsedRatio asynchronously
+	go func() {
+		gasUsedRatioCh <- convertFloat64SliceToArgUint64Slice(*gasUsedRatio)
+	}()
+
+	// Process reward asynchronously
+	go func() {
+		rewardCh <- convertToArgUint64SliceSlice(*reward)
+	}()
+
+	// Wait for the processed slices from goroutines
+	baseFeePerGasResult := <-baseFeePerGasCh
+	gasUsedRatioResult := <-gasUsedRatioCh
+	rewardResult := <-rewardCh
+
+	result := &feeHistoryResult{
+		OldestBlock:   argUint64(*oldestBlock),
+		BaseFeePerGas: baseFeePerGasResult,
+		GasUsedRatio:  gasUsedRatioResult,
+		Reward:        rewardResult,
+	}
+
+	return result, nil
+}
+
+func convertToArgUint64Slice(slice []uint64) []argUint64 {
+	argSlice := make([]argUint64, len(slice))
+	for i, value := range slice {
+		argSlice[i] = argUint64(value)
+	}
+
+	return argSlice
+}
+
+func convertToArgUint64SliceSlice(slice [][]uint64) [][]argUint64 {
+	argSlice := make([][]argUint64, len(slice))
+	for i, value := range slice {
+		argSlice[i] = convertToArgUint64Slice(value)
+	}
+
+	return argSlice
+}
+
+func convertFloat64SliceToArgUint64Slice(slice []float64) []argUint64 {
+	argSlice := make([]argUint64, len(slice))
+	for i, value := range slice {
+		argSlice[i] = argUint64(value)
+	}
+
+	return argSlice
+}
```

Comment:
I suppose it is better to move these helper functions into `jsonrpc/types.go`. Or maybe we could have even avoided it if `gasPrice.FeeHistory` was returning the correct types?

WDYT?

Reply:
moved to `jsonrpc/types.go` for time being.
I can either make `gasPrice.FeeHistory` return a `feeHistory` result or have it return values as `argUint64`s
Let me know what you think is better - cleaned it up for time being.

Reply:
> I can either make gasPrice.FeeHistory return a feeHistory result or have it return values as argUint64s before packing and returning from eth_endpoint.go

Thinking out loud: if we are about to be consistent with other JSON RPC API functions out there, then we should probably keep the current approach (that is converting it to those derived types only after the algorithm is executed and just before returning a response). Therefore I'd say we are good with the current approach.